### ISSUE TITLE
PHP-Auth version?

### ISSUE THREAD

**Issue by Paulo--M--** [delight-im/PHP-Auth#228]
Hello,
Is there a quick and simple way to know which version of PHP-Auth I'm using? Programmatically, that is.
Thanks.

**Comment by ocram** [comment]
You can use

```bash
$ composer show
```

in your project root folder, or more specifically

```bash
$ composer show delight-im/auth
```

which does (also) provide programmatic access. Is that sufficient?

Or do you want programmatic access from PHP? Could you describe the use case?

**Comment by Paulo--M--** [comment]
I meant programmatic from PHP. Specifically, I was looking at ways in which I could output the version of PHPAuth in use on sort of 'credits page' that lists various packages that are making my life easier :-) I could, of course, ask composer and hard-code the version number somewhere -- but I thought that it would be more elegant to get PHP to ask PHPAuth directly, if you know what I mean...

**Comment by ocram** [comment]
Thanks! Except for wrapping the above command in `exec` or the like, which probably wasn’t what you had hoped for, there is no such feature at the moment. Sorry!

We could certainly add a `Version` class in the base namespace in `src` that is updated with every new release. You could then use `Version::MAJOR`, `Version::MINOR`, `Version::PATCH`, or `Version::FULL` in your code.

But with regard to your example of a credits page, are you sure the version information is actually needed? I mean, it’s absolutely fine – and it’s also what most sites do – to only list the name of the software or library, not the exact version used. For security purposes, and this is even more relevant with an authentication library, you also wouldn’t ever want to tell potential attackers the exact version information. If there ever *was* a security vulnerability, the attackers could use a web search or web scraping to find sites using the exact vulnerable version.

Furthermore, you probably want to list other pieces of software on your credits page. Do they all offer a compatible system for retrieving version information, or would every single project need to be handled differently, which doesn’t really make things easier?

In the end, perhaps it’s really easier to query

```bash
$ composer show
```

for information on *all* libraries at once from time to time. What do you think?

**Comment by Paulo--M--** [comment]
I agree with your comments completely. I read your very informative note above at the time but somehow never got around to thanking you. Please accept my apologies for this rude behaviour.

**Comment by ocram** [comment]
No problem, thank you!
